### PR TITLE
2. Validate final score if game status is finish

### LINKED ISSUE #22: 2. Validate final score if game status is finish:
- score vector size 20 if no strike or spare
- score vector size 21 if spare at last run
- score vector size 22 if strike at last run

### PR DESCRIPTION
**Pull request by Kasperice**
Resolves: #22

### CODE REVIEW

File: bowling/player.cpp [hunk 1]
```diff
@@ -10,8 +10,9 @@ void Player::setScore(int score) {
 
 void Player::countScore(std::vector<std::pair<int, int>> score) {
     int result = 0;
+    int lastNormalRound = 9;
```
Comment (arafalski) [inline bowling/player.cpp]:
```suggestion
    constexpr int lastNormalRound = 9;
    
```

**Review by Vicx95 — DISMISSED** [review]

**Review by Vicx95 — DISMISSED** [review]

File: bowling/game.cpp [hunk 2]
```diff
@@ -1 +1,26 @@
...
+        isFinished_ = true;
+    } else if ((score[lastNormalRound].first == 10) && (score[lastNormalRound + 1].first != 10) &&
+               (score.size() == 11)) {
+        isFinished_ = true;
+    } else if ((score[lastNormalRound].first == 10) && (score[lastNormalRound + 1].first == 10) &&
+               (score.size() == 12)) {
+        isFinished_ = true;
+    } else {
+        isFinished_ = false;
+    }
```
Comment (mateusz-talma) [inline bowling/game.cpp]: Maybe this code will be more readable?

```cpp
if (score.size() == 10) {
    isFinished_ = checkEndConditionFor10Turns(score);
}
else if (score.size() == 11) {
    isFinished_ = checkEndConditionFor11Turns(score);
}
...

```

**Review by mateusz-talma — COMMENTED** [review]
Good code. Just look at my proposal for checkGameProgress

File: bowling/game.cpp (hunk 2 above)
Comment (Kasperice) [reply bowling/game.cpp]: I changed it, it is better. Thanks :)

File: bowling/game.hpp [hunk 3]
```diff
@@ -1,6 +1,17 @@
...
 public:
+    void checkGameProgress(std::vector<std::pair<int, int>> score);
+    bool getIsFinished() const;
     static void printHelp();
+
+private:
+    bool isFinished_;
+    bool checkEndConditionFor10Turns(std::vector<std::pair<int, int>> score);
+    bool checkEndConditionFor11Turns(std::vector<std::pair<int, int>> score);
+    bool checkEndConditionFor12Turns(std::vector<std::pair<int, int>> score);
```
Comment (mateusz-talma) [inline bowling/game.hpp]: I propose pass the vector by const reference to avoid copying :)

**Review by arafalski — COMMENTED** [review]
Please resolve conflicts

File: bowling/game.hpp (hunk 3 above)
Comment (Kasperice) [reply bowling/game.hpp]: Done

**Comment by Kasperice** [comment]
Conflicts resolved

**Review by arafalski — APPROVED** [review]

**Review by Vicx95 — APPROVED** [review]

**Pull request merged**